src/js/pyodide.ts

```diff
@@ -99,6 +109,14 @@ export async function loadPyodide(
      * Default: ```${indexURL}/pyodide-lock.json```
      */
     lockFileURL?: string;
+    /**
+     * TODO(now) fill this in
+     */
+    lockFileContents?: Promise<Lockfile | string>;
```

Comment:
How about accepting both `Promise<Lockfile | string>` and `Lockfile | string`? I know it is for optimization, but some people might want to preload lockfile, and feel weird to wrap it again with `Promise.resolve()`.

Reply:
Agreed - in my JupyterLite deployment I have a custom plugin that allows reading configs from URL query params, which I currently use to allow overriding the lock file URL. With this PR, I'd want to allow supplying the lock file as a string directly (and since it goes through a generic extension I could only supply the string but not a promise)

Reply:
Done.